transfermethodui/src/main/java/com/hyperwallet/android/ui/transfermethod/HyperwalletTransferMethodUi.java

```diff
@@ -58,6 +70,53 @@ public static synchronized HyperwalletTransferMethodUi getInstance(
         return sInstance;
     }
 
+    /**
+     * Returns an instance that will also initialize the Insights library.
+     *
+     * @param context                     A Context of the application consuming this Intent.
+     * @param authenticationTokenProvider An implementation of the {@link HyperwalletAuthenticationTokenProvider}
+     * @return singleton instance of HyperwalletTransferMethodUi
+     */
+    public static synchronized HyperwalletTransferMethodUi getInstance(@NonNull final Context context,
+            @NonNull final HyperwalletAuthenticationTokenProvider authenticationTokenProvider) {
+        if (sInstance == null) {
+            sInstance = new HyperwalletTransferMethodUi();
+        }
+        Executor executor = Executors.newSingleThreadExecutor();
+        executor.execute(new Runnable() {
+            @Override
+            public void run() {
+                Hyperwallet.getInstance(authenticationTokenProvider, new HyperwalletListener<Configuration>() {
+
+                    @Override
+                    public void onSuccess(@Nullable Configuration result) {
+                        if (result != null) {
+                            // TODO check if BuildConfig.DEBUG ?
+                            String environment = context.getString(R.string.environment);
+                            String sdkVersion = BuildConfig.VERSION_NAME;
+
+                            HyperwalletInsight.getInstance().initializeInsight(context, environment,
+                                    result.getProgramToken(), sdkVersion, result.getInsightApiUrl(),
+                                    result.getUserToken());
+                        }
+                    }
+
+                    @Override
+                    public void onFailure(HyperwalletException exception) {
+                        // TODO unsure what to log here
+                        final String logMessage = MessageFormat.format(
```

Comment:
do nothing, we are not logging or something, if this Insight call is not successful, we will try to initialize it again in the next Insight invocation